src/graph/transform/cuda/knn.cu

```diff
@@ -22,6 +23,14 @@
 namespace dgl {
 namespace transform {
 namespace impl {
+
+template <typename Type>
+static __host__ __device__ std::enable_if_t<std::is_unsigned_v<Type>, Type>
```

Comment:
`std::is_unsigned_v<>` is available since c++17 while minimum requirement in DGL is c++14. let's use `std::is_unsigned<>::value` instead?